x-pack/plugins/maps/public/classes/styles/vector/components/data_mapping/percentiles_form.tsx

```diff
@@ -0,0 +1,148 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import React, { ChangeEvent, Component } from 'react';
+import { EuiButton, EuiFieldNumber, EuiFormRow, EuiSpacer, EuiTextAlign } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { RowActionButtons } from '../row_action_buttons';
+
+interface Props {
+  initialPercentiles: number[];
+  onChange: (percentiles: number[]) => void;
+}
+
+interface State {
+  percentiles: Array<number | string>;
+}
+
+function isInvalidPercentile(percentile: unknown) {
+  if (typeof percentile !== 'number') {
+    return true;
+  }
+
+  return percentile <= 0 || percentile >= 100;
+}
+
+export class PercentilesForm extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      percentiles: props.initialPercentiles,
+    };
+  }
+
+  _onSubmit = () => {
+    this.props.onChange(this.state.percentiles as number[]);
+  };
+
+  render() {
+    const rows = this.state.percentiles.map((percentile: number | string, index: number) => {
+      const onAdd = () => {
+        let newPercentile: number | string = '';
+        if (typeof percentile === 'number') {
+          let delta = 1;
+          if (index === this.state.percentiles.length - 1) {
+            // Adding row to end of list.
+            if (index !== 0) {
+              const prevPercentile = this.state.percentiles[index - 1];
+              if (typeof prevPercentile === 'number') {
+                delta = percentile - prevPercentile;
+              }
+            }
+          } else {
+            // Adding row in middle of list.
+            const nextPercentile = this.state.percentiles[index + 1];
+            if (typeof nextPercentile === 'number') {
+              delta = (nextPercentile - percentile) / 2;
+            }
+          }
+          newPercentile = percentile + delta;
+          if (newPercentile >= 100) {
+            newPercentile = 99;
+          }
+        }
+
+        const percentiles = [
+          ...this.state.percentiles.slice(0, index + 1),
+          newPercentile,
+          ...this.state.percentiles.slice(index + 1),
+        ];
+        this.setState({ percentiles });
+      };
+
+      const onRemove = () => {
+        const percentiles =
+          this.state.percentiles.length === 1
+            ? this.state.percentiles
+            : [
+                ...this.state.percentiles.slice(0, index),
+                ...this.state.percentiles.slice(index + 1),
+              ];
+        this.setState({ percentiles });
+      };
+
+      const onPercentileChange = (event: ChangeEvent<HTMLInputElement>) => {
+        const sanitizedValue = parseFloat(event.target.value);
+        const percentiles = [...this.state.percentiles];
+        percentiles[index] = isNaN(sanitizedValue) ? '' : sanitizedValue;
+        this.setState({ percentiles });
+      };
+
```

Comment:
consider adding validation check so there are no dupes. If not, ES will throw an error.

e.g.
```
GET kibana_sample_data_logs/_search
{
  "size": 0,
  "aggs": {
    "bytes_percentiles": {
      "percentiles": {
        "percents": [
          0,
          50,
          75,
          82.5,
          90,
          90,
          95
        ],
        "field": "bytes"
      }
    }
  },
  "stored_fields": [
    "*"
  ],
  "script_fields": {
    "hour_of_day": {
      "script": {
        "source": "doc['timestamp'].value.getHour()",
        "lang": "painless"
      }
    }
  },
  "docvalue_fields": [
    {
      "field": "@timestamp",
      "format": "date_time"
    },
    {
      "field": "timestamp",
      "format": "date_time"
    },
    {
      "field": "utc_time",
      "format": "date_time"
    }
  ],
  "_source": {
    "excludes": []
  },
  "query": {
    "bool": {
      "must": [],
      "filter": [
        {
          "range": {
            "timestamp": {
              "gte": "2019-12-01T21:01:43.175Z",
              "lte": "2020-12-01T21:01:43.175Z",
              "format": "strict_date_optional_time"
            }
          }
        }
      ],
      "should": [],
      "must_not": []
    }
  }
}
```

will throw.